clang/include/clang/AST/Type.h

```diff
@@ -607,13 +799,17 @@ class Qualifiers {
 
   void Profile(llvm::FoldingSetNodeID &ID) const {
     ID.AddInteger(Mask);
+    PtrAuth.Profile(ID);
   }
 
 private:
   // bits:     |0 1 2|3|4 .. 5|6  ..  8|9   ...   31|
   //           |C R V|U|GCAttr|Lifetime|AddressSpace|
   uint32_t Mask = 0;
 
+  PointerAuthQualifier PtrAuth;
```

Comment:
This doubles the size of `Qualifiers` which is passed by value all over the place. I don't expect significant performance issues from this (though we do still support 32-bit builds of Clang), but do you have ideas on how this impacts compile time performance and memory pressure? (Changes in the type system can have outsized impact in surprising places.)

It may be worth putting a branch up on https://llvm-compile-time-tracker.com/ to see how this impacts things.

Reply:
That was used for at least several months in downstream and I'm not aware of compile time issues it caused. But, as you've mentioned, the changes can lead to undesirable impact in surprising places, so it's better to test it anyway. I'll do that and let you know about results.

Do I get it right that I need to contact nikic to be able to test branches from my fork (https://llvm-compile-time-tracker.com/about.php)? When requesting smth on the compare page (https://llvm-compile-time-tracker.com/compare.php), only commits and branches from this repository seem to be allowed (while I want to use the branch from my fork).

Reply:
That's correct, you'd contact @nikic to get your fork up there.